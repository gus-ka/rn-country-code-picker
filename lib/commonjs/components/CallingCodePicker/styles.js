"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("react");
var _reactNative = require("react-native");
var _theme = require("../../theme");
const shadowStyle = _reactNative.StyleSheet.create({
  listContainer: _reactNative.Platform.OS === 'ios' ? {
    shadowColor: _theme.colors.black,
    shadowOpacity: 0.5,
    shadowRadius: 5,
    shadowOffset: {
      height: 2,
      width: 2
    }
  } : {
    elevation: 3,
    overflow: 'hidden'
  }
});
const useStyles = y => {
  const style = (0, _react.useMemo)(() => _reactNative.StyleSheet.create({
    keyboardAvoidingView: {
      position: 'absolute',
      alignSelf: 'center',
      top: y,
      width: '90%'
    },
    listContainer: {
      ...shadowStyle.listContainer,
      width: '100%',
      height: _theme.MODAL_SIZE,
      backgroundColor: _theme.colors.white,
      padding: _theme.spacing.s,
      borderRadius: _theme.borderRadius.m,
      justifyContent: 'center',
      marginTop: _theme.spacing.s
    },
    dismissButton: {
      flex: 1,
      width: '100%'
    },
    modalChild: {
      flex: 1
    }
  }), [y]);
  return style;
};
var _default = useStyles;
exports.default = _default;
//# sourceMappingURL=styles.js.map