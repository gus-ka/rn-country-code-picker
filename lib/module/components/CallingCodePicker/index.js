import React, { createRef, useEffect, useState } from 'react';
import { Dimensions, FlatList, KeyboardAvoidingView, Modal, Platform, TouchableOpacity, View } from 'react-native';
import * as RNLocalize from 'react-native-localize';
import countries from '../../data/countries';
import { onLayoutToggle, sortData } from '../../functions/utility';
import { MODAL_SIZE, spacing } from '../../theme';
import ItemSeparator from '../ItemSeparator';
import PickerItem from '../PickerItem';
import PickerToggler from '../PickerToggler';
import Search from '../Search';
import useStyles from './styles';
const CallingCodePicker = _ref => {
  let {
    initialCountryCode,
    onValueChange,
    togglerContainerStyle,
    togglerLabelStyle,
    listContainerStyle,
    listStyle,
    pickerItemLabelStyle,
    searchInputStyle,
    pickerItemContainerStyle,
    style,
    isFlagVisible = true
  } = _ref;
  const [searchValue, setSearchValue] = useState('');
  const [isPickerOpen, setIsPickerOpen] = useState(false);
  const [countriesData, setCountriesData] = useState(sortData(countries, ''));
  const localCountryCode = countries.find(f => f.alpha2Code === RNLocalize.getCountry());
  const [selectedCountry, setSelectedCountry] = useState(initialCountryCode ? countries.find(f => f.alpha2Code === initialCountryCode) ?? localCountryCode : localCountryCode);
  const containerRef = /*#__PURE__*/createRef();
  const [toggleMeasure, setToggleMeasure] = useState();
  const [containerMeasure, setContainerMeasure] = useState();
  const styles = useStyles(calculateModalVerticalPosition());
  const handleCountrySelect = country => {
    setSelectedCountry(country);
    setIsPickerOpen(false);
    setSearchValue('');
  };
  useEffect(() => {
    let newCountries = countries.filter(s => s.callingCode === searchValue.trim() || s.alpha2Code.trim().toLowerCase().includes(searchValue.trim().toLowerCase()) || s.name.trim().toLowerCase().includes(searchValue.trim().toLowerCase()));
    newCountries = sortData(newCountries, searchValue);
    setCountriesData(newCountries);
  }, [searchValue]);
  const renderPickerItem = renderItem => {
    const {
      item: country
    } = renderItem;
    return /*#__PURE__*/React.createElement(PickerItem, {
      country: country,
      onCountrySelect: handleCountrySelect,
      textStyle: pickerItemLabelStyle,
      containerStyle: pickerItemContainerStyle
    });
  };
  function calculateModalVerticalPosition() {
    const screenHeight = Dimensions.get('window').height;
    const y = ((containerMeasure === null || containerMeasure === void 0 ? void 0 : containerMeasure.y) ?? 0) + ((toggleMeasure === null || toggleMeasure === void 0 ? void 0 : toggleMeasure.height) ?? 0);
    return y + MODAL_SIZE > screenHeight ? screenHeight - MODAL_SIZE - 2 * spacing.s : y;
  }
  useEffect(() => {
    onValueChange(selectedCountry === null || selectedCountry === void 0 ? void 0 : selectedCountry.callingCode);
  }, [selectedCountry, onValueChange]);
  function dismissPicker() {
    setIsPickerOpen(false);
    setSearchValue('');
  }
  return /*#__PURE__*/React.createElement(View, {
    style: style,
    ref: containerRef,
    onLayout: () => onLayoutToggle(containerRef, measure => setContainerMeasure(measure))
  }, /*#__PURE__*/React.createElement(PickerToggler, {
    flag: selectedCountry === null || selectedCountry === void 0 ? void 0 : selectedCountry.flag,
    selectedCode: selectedCountry === null || selectedCountry === void 0 ? void 0 : selectedCountry.callingCode,
    isPickerOpen: isPickerOpen,
    onPickerToggle: setIsPickerOpen,
    containerStyle: togglerContainerStyle,
    textStyle: togglerLabelStyle,
    onLayout: measure => setToggleMeasure(measure),
    isFlagVisible: isFlagVisible
  }), /*#__PURE__*/React.createElement(Modal, {
    visible: isPickerOpen,
    transparent: true,
    animationType: "fade"
  }, /*#__PURE__*/React.createElement(View, {
    style: styles.modalChild
  }, /*#__PURE__*/React.createElement(TouchableOpacity, {
    style: styles.dismissButton,
    onPress: dismissPicker
  }), /*#__PURE__*/React.createElement(KeyboardAvoidingView, {
    behavior: "position",
    style: styles.keyboardAvoidingView,
    keyboardVerticalOffset: Platform.OS === 'ios' ? 10 : -60
  }, /*#__PURE__*/React.createElement(View, {
    style: listContainerStyle ?? styles.listContainer
  }, /*#__PURE__*/React.createElement(Search, {
    value: searchValue,
    onChangeText: setSearchValue,
    onClearInput: () => setSearchValue(''),
    inputStyle: searchInputStyle
  }), /*#__PURE__*/React.createElement(FlatList, {
    data: countriesData,
    renderItem: renderPickerItem,
    keyExtractor: (_, index) => index.toString(),
    showsVerticalScrollIndicator: false,
    style: listStyle,
    ItemSeparatorComponent: ItemSeparator,
    keyboardShouldPersistTaps: "always"
  }))))));
};
export default CallingCodePicker;
//# sourceMappingURL=index.js.map