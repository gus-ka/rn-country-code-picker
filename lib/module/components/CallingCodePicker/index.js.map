{"version":3,"names":["React","createRef","useEffect","useState","Dimensions","FlatList","KeyboardAvoidingView","Modal","Platform","TouchableOpacity","View","RNLocalize","countries","onLayoutToggle","sortData","MODAL_SIZE","spacing","ItemSeparator","PickerItem","PickerToggler","Search","useStyles","CallingCodePicker","_ref","initialCountryCode","onValueChange","togglerContainerStyle","togglerLabelStyle","listContainerStyle","listStyle","pickerItemLabelStyle","searchInputStyle","pickerItemContainerStyle","style","isFlagVisible","searchValue","setSearchValue","isPickerOpen","setIsPickerOpen","countriesData","setCountriesData","localCountryCode","find","f","alpha2Code","getCountry","selectedCountry","setSelectedCountry","containerRef","toggleMeasure","setToggleMeasure","containerMeasure","setContainerMeasure","styles","calculateModalVerticalPosition","handleCountrySelect","country","newCountries","filter","s","callingCode","trim","toLowerCase","includes","name","renderPickerItem","renderItem","item","createElement","onCountrySelect","textStyle","containerStyle","screenHeight","get","height","y","dismissPicker","ref","onLayout","measure","flag","selectedCode","onPickerToggle","visible","transparent","animationType","modalChild","dismissButton","onPress","behavior","keyboardAvoidingView","keyboardVerticalOffset","OS","listContainer","value","onChangeText","onClearInput","inputStyle","data","keyExtractor","_","index","toString","showsVerticalScrollIndicator","ItemSeparatorComponent","keyboardShouldPersistTaps"],"sources":["index.tsx"],"sourcesContent":["import React, { createRef, useEffect, useState } from 'react';\nimport {\n  Dimensions,\n  FlatList,\n  KeyboardAvoidingView,\n  ListRenderItemInfo,\n  Modal,\n  Platform,\n  TouchableOpacity,\n  View,\n} from 'react-native';\nimport * as RNLocalize from 'react-native-localize';\nimport countries from '../../data/countries';\nimport { onLayoutToggle, sortData } from '../../functions/utility';\nimport { MODAL_SIZE, spacing } from '../../theme';\nimport { ICallingCodePickerProps, ICountry, IItemMeasure } from '../../types';\nimport ItemSeparator from '../ItemSeparator';\nimport PickerItem from '../PickerItem';\nimport PickerToggler from '../PickerToggler';\nimport Search from '../Search';\nimport useStyles from './styles';\n\nconst CallingCodePicker: React.FC<ICallingCodePickerProps> = ({\n  initialCountryCode,\n  onValueChange,\n  togglerContainerStyle,\n  togglerLabelStyle,\n  listContainerStyle,\n  listStyle,\n  pickerItemLabelStyle,\n  searchInputStyle,\n  pickerItemContainerStyle,\n  style,\n  isFlagVisible = true,\n}) => {\n  const [searchValue, setSearchValue] = useState<string>('');\n  const [isPickerOpen, setIsPickerOpen] = useState<boolean>(false);\n  const [countriesData, setCountriesData] = useState<ICountry[]>(sortData(countries, ''));\n  const localCountryCode = countries.find(f => f.alpha2Code === RNLocalize.getCountry());\n  const [selectedCountry, setSelectedCountry] = useState<ICountry | undefined>(\n    initialCountryCode\n      ? countries.find(f => f.alpha2Code === initialCountryCode) ?? localCountryCode\n      : localCountryCode,\n  );\n  const containerRef = createRef<View>();\n  const [toggleMeasure, setToggleMeasure] = useState<IItemMeasure>();\n  const [containerMeasure, setContainerMeasure] = useState<IItemMeasure>();\n  const styles = useStyles(calculateModalVerticalPosition());\n\n  const handleCountrySelect = (country: ICountry) => {\n    setSelectedCountry(country);\n    setIsPickerOpen(false);\n    setSearchValue('');\n  };\n\n  useEffect(() => {\n    let newCountries = countries.filter(\n      s =>\n        s.callingCode === searchValue.trim() ||\n        s.alpha2Code.trim().toLowerCase().includes(searchValue.trim().toLowerCase()) ||\n        s.name.trim().toLowerCase().includes(searchValue.trim().toLowerCase()),\n    );\n\n    newCountries = sortData(newCountries, searchValue);\n\n    setCountriesData(newCountries);\n  }, [searchValue]);\n\n  const renderPickerItem = (renderItem: ListRenderItemInfo<ICountry>) => {\n    const { item: country } = renderItem;\n    return (\n      <PickerItem\n        country={country}\n        onCountrySelect={handleCountrySelect}\n        textStyle={pickerItemLabelStyle}\n        containerStyle={pickerItemContainerStyle}\n      />\n    );\n  };\n\n  function calculateModalVerticalPosition() {\n    const screenHeight = Dimensions.get('window').height;\n    const y = (containerMeasure?.y ?? 0) + (toggleMeasure?.height ?? 0);\n\n    return y + MODAL_SIZE > screenHeight ? screenHeight - MODAL_SIZE - 2 * spacing.s : y;\n  }\n\n  useEffect(() => {\n    onValueChange(selectedCountry?.callingCode);\n  }, [selectedCountry, onValueChange]);\n\n  function dismissPicker() {\n    setIsPickerOpen(false);\n    setSearchValue('');\n  }\n\n  return (\n    <View\n      style={style}\n      ref={containerRef}\n      onLayout={() => onLayoutToggle(containerRef, measure => setContainerMeasure(measure))}\n    >\n      <PickerToggler\n        flag={selectedCountry?.flag}\n        selectedCode={selectedCountry?.callingCode}\n        isPickerOpen={isPickerOpen}\n        onPickerToggle={setIsPickerOpen}\n        containerStyle={togglerContainerStyle}\n        textStyle={togglerLabelStyle}\n        onLayout={measure => setToggleMeasure(measure)}\n        isFlagVisible={isFlagVisible}\n      />\n      <Modal visible={isPickerOpen} transparent animationType=\"fade\">\n        <View style={styles.modalChild}>\n          <TouchableOpacity style={styles.dismissButton} onPress={dismissPicker} />\n          <KeyboardAvoidingView\n            behavior=\"position\"\n            style={styles.keyboardAvoidingView}\n            keyboardVerticalOffset={Platform.OS === 'ios' ? 10 : -60}\n          >\n            <View style={listContainerStyle ?? styles.listContainer}>\n              <Search\n                value={searchValue}\n                onChangeText={setSearchValue}\n                onClearInput={() => setSearchValue('')}\n                inputStyle={searchInputStyle}\n              />\n              <FlatList\n                data={countriesData}\n                renderItem={renderPickerItem}\n                keyExtractor={(_, index) => index.toString()}\n                showsVerticalScrollIndicator={false}\n                style={listStyle}\n                ItemSeparatorComponent={ItemSeparator}\n                keyboardShouldPersistTaps=\"always\"\n              />\n            </View>\n          </KeyboardAvoidingView>\n        </View>\n      </Modal>\n    </View>\n  );\n};\n\nexport default CallingCodePicker;\n"],"mappings":"AAAA,OAAOA,KAAK,IAAIC,SAAS,EAAEC,SAAS,EAAEC,QAAQ,QAAQ,OAAO;AAC7D,SACEC,UAAU,EACVC,QAAQ,EACRC,oBAAoB,EAEpBC,KAAK,EACLC,QAAQ,EACRC,gBAAgB,EAChBC,IAAI,QACC,cAAc;AACrB,OAAO,KAAKC,UAAU,MAAM,uBAAuB;AACnD,OAAOC,SAAS,MAAM,sBAAsB;AAC5C,SAASC,cAAc,EAAEC,QAAQ,QAAQ,yBAAyB;AAClE,SAASC,UAAU,EAAEC,OAAO,QAAQ,aAAa;AAEjD,OAAOC,aAAa,MAAM,kBAAkB;AAC5C,OAAOC,UAAU,MAAM,eAAe;AACtC,OAAOC,aAAa,MAAM,kBAAkB;AAC5C,OAAOC,MAAM,MAAM,WAAW;AAC9B,OAAOC,SAAS,MAAM,UAAU;AAEhC,MAAMC,iBAAoD,GAAGC,IAAA,IAYvD;EAAA,IAZwD;IAC5DC,kBAAkB;IAClBC,aAAa;IACbC,qBAAqB;IACrBC,iBAAiB;IACjBC,kBAAkB;IAClBC,SAAS;IACTC,oBAAoB;IACpBC,gBAAgB;IAChBC,wBAAwB;IACxBC,KAAK;IACLC,aAAa,GAAG;EAClB,CAAC,GAAAX,IAAA;EACC,MAAM,CAACY,WAAW,EAAEC,cAAc,CAAC,GAAGjC,QAAQ,CAAS,EAAE,CAAC;EAC1D,MAAM,CAACkC,YAAY,EAAEC,eAAe,CAAC,GAAGnC,QAAQ,CAAU,KAAK,CAAC;EAChE,MAAM,CAACoC,aAAa,EAAEC,gBAAgB,CAAC,GAAGrC,QAAQ,CAAaW,QAAQ,CAACF,SAAS,EAAE,EAAE,CAAC,CAAC;EACvF,MAAM6B,gBAAgB,GAAG7B,SAAS,CAAC8B,IAAI,CAACC,CAAC,IAAIA,CAAC,CAACC,UAAU,KAAKjC,UAAU,CAACkC,UAAU,CAAC,CAAC,CAAC;EACtF,MAAM,CAACC,eAAe,EAAEC,kBAAkB,CAAC,GAAG5C,QAAQ,CACpDqB,kBAAkB,GACdZ,SAAS,CAAC8B,IAAI,CAACC,CAAC,IAAIA,CAAC,CAACC,UAAU,KAAKpB,kBAAkB,CAAC,IAAIiB,gBAAgB,GAC5EA,gBACN,CAAC;EACD,MAAMO,YAAY,gBAAG/C,SAAS,CAAO,CAAC;EACtC,MAAM,CAACgD,aAAa,EAAEC,gBAAgB,CAAC,GAAG/C,QAAQ,CAAe,CAAC;EAClE,MAAM,CAACgD,gBAAgB,EAAEC,mBAAmB,CAAC,GAAGjD,QAAQ,CAAe,CAAC;EACxE,MAAMkD,MAAM,GAAGhC,SAAS,CAACiC,8BAA8B,CAAC,CAAC,CAAC;EAE1D,MAAMC,mBAAmB,GAAIC,OAAiB,IAAK;IACjDT,kBAAkB,CAACS,OAAO,CAAC;IAC3BlB,eAAe,CAAC,KAAK,CAAC;IACtBF,cAAc,CAAC,EAAE,CAAC;EACpB,CAAC;EAEDlC,SAAS,CAAC,MAAM;IACd,IAAIuD,YAAY,GAAG7C,SAAS,CAAC8C,MAAM,CACjCC,CAAC,IACCA,CAAC,CAACC,WAAW,KAAKzB,WAAW,CAAC0B,IAAI,CAAC,CAAC,IACpCF,CAAC,CAACf,UAAU,CAACiB,IAAI,CAAC,CAAC,CAACC,WAAW,CAAC,CAAC,CAACC,QAAQ,CAAC5B,WAAW,CAAC0B,IAAI,CAAC,CAAC,CAACC,WAAW,CAAC,CAAC,CAAC,IAC5EH,CAAC,CAACK,IAAI,CAACH,IAAI,CAAC,CAAC,CAACC,WAAW,CAAC,CAAC,CAACC,QAAQ,CAAC5B,WAAW,CAAC0B,IAAI,CAAC,CAAC,CAACC,WAAW,CAAC,CAAC,CACzE,CAAC;IAEDL,YAAY,GAAG3C,QAAQ,CAAC2C,YAAY,EAAEtB,WAAW,CAAC;IAElDK,gBAAgB,CAACiB,YAAY,CAAC;EAChC,CAAC,EAAE,CAACtB,WAAW,CAAC,CAAC;EAEjB,MAAM8B,gBAAgB,GAAIC,UAAwC,IAAK;IACrE,MAAM;MAAEC,IAAI,EAAEX;IAAQ,CAAC,GAAGU,UAAU;IACpC,oBACElE,KAAA,CAAAoE,aAAA,CAAClD,UAAU;MACTsC,OAAO,EAAEA,OAAQ;MACjBa,eAAe,EAAEd,mBAAoB;MACrCe,SAAS,EAAExC,oBAAqB;MAChCyC,cAAc,EAAEvC;IAAyB,CAC1C,CAAC;EAEN,CAAC;EAED,SAASsB,8BAA8BA,CAAA,EAAG;IACxC,MAAMkB,YAAY,GAAGpE,UAAU,CAACqE,GAAG,CAAC,QAAQ,CAAC,CAACC,MAAM;IACpD,MAAMC,CAAC,GAAG,CAAC,CAAAxB,gBAAgB,aAAhBA,gBAAgB,uBAAhBA,gBAAgB,CAAEwB,CAAC,KAAI,CAAC,KAAK,CAAA1B,aAAa,aAAbA,aAAa,uBAAbA,aAAa,CAAEyB,MAAM,KAAI,CAAC,CAAC;IAEnE,OAAOC,CAAC,GAAG5D,UAAU,GAAGyD,YAAY,GAAGA,YAAY,GAAGzD,UAAU,GAAG,CAAC,GAAGC,OAAO,CAAC2C,CAAC,GAAGgB,CAAC;EACtF;EAEAzE,SAAS,CAAC,MAAM;IACduB,aAAa,CAACqB,eAAe,aAAfA,eAAe,uBAAfA,eAAe,CAAEc,WAAW,CAAC;EAC7C,CAAC,EAAE,CAACd,eAAe,EAAErB,aAAa,CAAC,CAAC;EAEpC,SAASmD,aAAaA,CAAA,EAAG;IACvBtC,eAAe,CAAC,KAAK,CAAC;IACtBF,cAAc,CAAC,EAAE,CAAC;EACpB;EAEA,oBACEpC,KAAA,CAAAoE,aAAA,CAAC1D,IAAI;IACHuB,KAAK,EAAEA,KAAM;IACb4C,GAAG,EAAE7B,YAAa;IAClB8B,QAAQ,EAAEA,CAAA,KAAMjE,cAAc,CAACmC,YAAY,EAAE+B,OAAO,IAAI3B,mBAAmB,CAAC2B,OAAO,CAAC;EAAE,gBAEtF/E,KAAA,CAAAoE,aAAA,CAACjD,aAAa;IACZ6D,IAAI,EAAElC,eAAe,aAAfA,eAAe,uBAAfA,eAAe,CAAEkC,IAAK;IAC5BC,YAAY,EAAEnC,eAAe,aAAfA,eAAe,uBAAfA,eAAe,CAAEc,WAAY;IAC3CvB,YAAY,EAAEA,YAAa;IAC3B6C,cAAc,EAAE5C,eAAgB;IAChCiC,cAAc,EAAE7C,qBAAsB;IACtC4C,SAAS,EAAE3C,iBAAkB;IAC7BmD,QAAQ,EAAEC,OAAO,IAAI7B,gBAAgB,CAAC6B,OAAO,CAAE;IAC/C7C,aAAa,EAAEA;EAAc,CAC9B,CAAC,eACFlC,KAAA,CAAAoE,aAAA,CAAC7D,KAAK;IAAC4E,OAAO,EAAE9C,YAAa;IAAC+C,WAAW;IAACC,aAAa,EAAC;EAAM,gBAC5DrF,KAAA,CAAAoE,aAAA,CAAC1D,IAAI;IAACuB,KAAK,EAAEoB,MAAM,CAACiC;EAAW,gBAC7BtF,KAAA,CAAAoE,aAAA,CAAC3D,gBAAgB;IAACwB,KAAK,EAAEoB,MAAM,CAACkC,aAAc;IAACC,OAAO,EAAEZ;EAAc,CAAE,CAAC,eACzE5E,KAAA,CAAAoE,aAAA,CAAC9D,oBAAoB;IACnBmF,QAAQ,EAAC,UAAU;IACnBxD,KAAK,EAAEoB,MAAM,CAACqC,oBAAqB;IACnCC,sBAAsB,EAAEnF,QAAQ,CAACoF,EAAE,KAAK,KAAK,GAAG,EAAE,GAAG,CAAC;EAAG,gBAEzD5F,KAAA,CAAAoE,aAAA,CAAC1D,IAAI;IAACuB,KAAK,EAAEL,kBAAkB,IAAIyB,MAAM,CAACwC;EAAc,gBACtD7F,KAAA,CAAAoE,aAAA,CAAChD,MAAM;IACL0E,KAAK,EAAE3D,WAAY;IACnB4D,YAAY,EAAE3D,cAAe;IAC7B4D,YAAY,EAAEA,CAAA,KAAM5D,cAAc,CAAC,EAAE,CAAE;IACvC6D,UAAU,EAAElE;EAAiB,CAC9B,CAAC,eACF/B,KAAA,CAAAoE,aAAA,CAAC/D,QAAQ;IACP6F,IAAI,EAAE3D,aAAc;IACpB2B,UAAU,EAAED,gBAAiB;IAC7BkC,YAAY,EAAEA,CAACC,CAAC,EAAEC,KAAK,KAAKA,KAAK,CAACC,QAAQ,CAAC,CAAE;IAC7CC,4BAA4B,EAAE,KAAM;IACpCtE,KAAK,EAAEJ,SAAU;IACjB2E,sBAAsB,EAAEvF,aAAc;IACtCwF,yBAAyB,EAAC;EAAQ,CACnC,CACG,CACc,CAClB,CACD,CACH,CAAC;AAEX,CAAC;AAED,eAAenF,iBAAiB"}