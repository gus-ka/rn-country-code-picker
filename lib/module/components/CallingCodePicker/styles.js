import { useMemo } from 'react';
import { Platform, StyleSheet } from 'react-native';
import { borderRadius, colors, MODAL_SIZE, spacing } from '../../theme';
const shadowStyle = StyleSheet.create({
  listContainer: Platform.OS === 'ios' ? {
    shadowColor: colors.black,
    shadowOpacity: 0.5,
    shadowRadius: 5,
    shadowOffset: {
      height: 2,
      width: 2
    }
  } : {
    elevation: 3,
    overflow: 'hidden'
  }
});
const useStyles = y => {
  const style = useMemo(() => StyleSheet.create({
    keyboardAvoidingView: {
      position: 'absolute',
      alignSelf: 'center',
      top: y,
      width: '90%'
    },
    listContainer: {
      ...shadowStyle.listContainer,
      width: '100%',
      height: MODAL_SIZE,
      backgroundColor: colors.white,
      padding: spacing.s,
      borderRadius: borderRadius.m,
      justifyContent: 'center',
      marginTop: spacing.s
    },
    dismissButton: {
      flex: 1,
      width: '100%'
    },
    modalChild: {
      flex: 1
    }
  }), [y]);
  return style;
};
export default useStyles;
//# sourceMappingURL=styles.js.map