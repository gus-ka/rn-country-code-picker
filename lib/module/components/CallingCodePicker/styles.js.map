{"version":3,"names":["useMemo","Platform","StyleSheet","borderRadius","colors","MODAL_SIZE","spacing","shadowStyle","create","listContainer","OS","shadowColor","black","shadowOpacity","shadowRadius","shadowOffset","height","width","elevation","overflow","useStyles","y","style","keyboardAvoidingView","position","alignSelf","top","backgroundColor","white","padding","s","m","justifyContent","marginTop","dismissButton","flex","modalChild"],"sources":["styles.ts"],"sourcesContent":["import { useMemo } from 'react';\nimport { Platform, StyleSheet } from 'react-native';\nimport { borderRadius, colors, MODAL_SIZE, spacing } from '../../theme';\n\nconst shadowStyle = StyleSheet.create({\n  listContainer:\n    Platform.OS === 'ios'\n      ? {\n          shadowColor: colors.black,\n          shadowOpacity: 0.5,\n          shadowRadius: 5,\n          shadowOffset: { height: 2, width: 2 },\n        }\n      : {\n          elevation: 3,\n          overflow: 'hidden',\n        },\n});\n\nconst useStyles = (y: number) => {\n  const style = useMemo(\n    () =>\n      StyleSheet.create({\n        keyboardAvoidingView: {\n          position: 'absolute',\n          alignSelf: 'center',\n          top: y,\n          width: '90%',\n        },\n        listContainer: {\n          ...shadowStyle.listContainer,\n          width: '100%',\n          height: MODAL_SIZE,\n          backgroundColor: colors.white,\n          padding: spacing.s,\n          borderRadius: borderRadius.m,\n          justifyContent: 'center',\n          marginTop: spacing.s,\n        },\n        dismissButton: { flex: 1, width: '100%' },\n        modalChild: { flex: 1 },\n      }),\n    [y],\n  );\n\n  return style;\n};\n\nexport default useStyles;\n"],"mappings":"AAAA,SAASA,OAAO,QAAQ,OAAO;AAC/B,SAASC,QAAQ,EAAEC,UAAU,QAAQ,cAAc;AACnD,SAASC,YAAY,EAAEC,MAAM,EAAEC,UAAU,EAAEC,OAAO,QAAQ,aAAa;AAEvE,MAAMC,WAAW,GAAGL,UAAU,CAACM,MAAM,CAAC;EACpCC,aAAa,EACXR,QAAQ,CAACS,EAAE,KAAK,KAAK,GACjB;IACEC,WAAW,EAAEP,MAAM,CAACQ,KAAK;IACzBC,aAAa,EAAE,GAAG;IAClBC,YAAY,EAAE,CAAC;IACfC,YAAY,EAAE;MAAEC,MAAM,EAAE,CAAC;MAAEC,KAAK,EAAE;IAAE;EACtC,CAAC,GACD;IACEC,SAAS,EAAE,CAAC;IACZC,QAAQ,EAAE;EACZ;AACR,CAAC,CAAC;AAEF,MAAMC,SAAS,GAAIC,CAAS,IAAK;EAC/B,MAAMC,KAAK,GAAGtB,OAAO,CACnB,MACEE,UAAU,CAACM,MAAM,CAAC;IAChBe,oBAAoB,EAAE;MACpBC,QAAQ,EAAE,UAAU;MACpBC,SAAS,EAAE,QAAQ;MACnBC,GAAG,EAAEL,CAAC;MACNJ,KAAK,EAAE;IACT,CAAC;IACDR,aAAa,EAAE;MACb,GAAGF,WAAW,CAACE,aAAa;MAC5BQ,KAAK,EAAE,MAAM;MACbD,MAAM,EAAEX,UAAU;MAClBsB,eAAe,EAAEvB,MAAM,CAACwB,KAAK;MAC7BC,OAAO,EAAEvB,OAAO,CAACwB,CAAC;MAClB3B,YAAY,EAAEA,YAAY,CAAC4B,CAAC;MAC5BC,cAAc,EAAE,QAAQ;MACxBC,SAAS,EAAE3B,OAAO,CAACwB;IACrB,CAAC;IACDI,aAAa,EAAE;MAAEC,IAAI,EAAE,CAAC;MAAElB,KAAK,EAAE;IAAO,CAAC;IACzCmB,UAAU,EAAE;MAAED,IAAI,EAAE;IAAE;EACxB,CAAC,CAAC,EACJ,CAACd,CAAC,CACJ,CAAC;EAED,OAAOC,KAAK;AACd,CAAC;AAED,eAAeF,SAAS"}