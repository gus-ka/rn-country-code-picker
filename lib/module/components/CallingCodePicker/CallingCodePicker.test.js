import React from 'react';
import { render } from '@testing-library/react-native';
import CallingCodePicker from './';
jest.mock('react-native-localize');
test('renders correctly', () => {
  const onValueChangeMockFn = jest.fn();
  const {
    toJSON
  } = render( /*#__PURE__*/React.createElement(CallingCodePicker, {
    onValueChange: onValueChangeMockFn
  }));
  const rendered = toJSON();
  expect(onValueChangeMockFn).toHaveBeenCalledWith('1');
  expect(rendered).toMatchSnapshot();
});
//# sourceMappingURL=CallingCodePicker.test.js.map