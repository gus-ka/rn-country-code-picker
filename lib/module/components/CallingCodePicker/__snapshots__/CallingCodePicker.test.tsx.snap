// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
>
  <View
    onLayout={[Function]}
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="toggler-button"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 4,
              "height": 30,
              "marginRight": 8,
              "width": 40,
            }
          }
        >
          <Image
            resizeMethod="scale"
            source={
              Object {
                "testUri": "../../../src/assets/flags/us.png",
              }
            }
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
            testID="flag-image"
          />
        </View>
        <Text
          style={
            Array [
              undefined,
              Object {
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
        >
          +1
           
        </Text>
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={18}
            bbWidth={18}
            color={4279308561}
            fill="none"
            focusable={false}
            height={18}
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke="currentColor"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 18,
                  "width": 18,
                },
              ]
            }
            tintColor={4279308561}
            vbHeight={24}
            vbWidth={24}
            width={18}
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                  "stroke",
                ]
              }
              stroke={
                Array [
                  2,
                ]
              }
            >
              <RNSVGPath
                d="M19 9l-7 7-7-7"
                propList={
                  Array [
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={2}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />
</View>
`;
